optuna/samplers/_tpe/sampler.py

```diff
@@ -324,14 +324,14 @@ def _split_observation_pairs(
         self, config_vals: List[Optional[float]], loss_vals: List[Tuple[float, float]]
     ) -> Tuple[np.ndarray, np.ndarray]:
 
-        config_vals = np.asarray(config_vals)
-        loss_vals = np.asarray(loss_vals, dtype=[("step", float), ("score", float)])
+        config_values = np.asarray(config_vals)
+        loss_values = np.asarray(loss_vals, dtype=[("step", float), ("score", float)])
```

Comment:
We can avoid introducing new variables by using `Union`. 
```suggestion
        config_vals: Union[List[Optional[float]], np.ndarray] = np.asarray(config_vals)
        loss_vals: Union[List[Optional[float]], np.ndarray] = np.asarray(loss_vals, dtype=[("step", float), ("score", float)])
```